### PR TITLE
add Closures README

### PR DESCRIPTION
**Pull request by tomhuhges**
I've created a short README on closures. Let me know what you think! ヽ(〃･ω･)ﾉ

### CODE REVIEW

File: JavaScript/Closures.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
+# Closures
+
+### Introduction
+
+Closures have a reputation for being one of JavaScript's most difficult concepts to grasp. However, if you're already comfortable with how JavaScript deals with functions and [scope][1], you're already halfway there.
```
Comment (lucymonie) [inline JavaScript/Closures.md:5]: Your scope link seems to be broken. Might want to fix.

File: JavaScript/Closures.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+next(); // returns 'b'
+next(); // returns 'c'
+next(); // returns 'd'
+...
+```
+
+---- 
+
+### Related
+
```
Comment (lucymonie) [inline JavaScript/Closures.md:148]: All the links on scope and hoisting are broken.

File: JavaScript/Closures.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+[JavaScript Scope: A Beginners Perspective][4]  
+[Variable scope and hoisting][5]
+[Hoisting][6]
+
```
Comment (alwinsol) [inline JavaScript/Closures.md:152]: I wrote an article on closures in today I learned https://github.com/codingforeveryone/today-i-learned/blob/master/oojs/closures.md

File: JavaScript/Closures.md (hunk 2 above)
Comment (tomhuhges) [inline JavaScript/Closures.md:148]: thanks - have fixed with absolute URLs

File: JavaScript/Closures.md (hunk 3 above)
Comment (tomhuhges) [inline JavaScript/Closures.md:152]: have added to the resources list!

**Comment by dgrcode** [comment]
I have a StackOverflow article that I found very insightful. I think it could be a good resource to add the the list, if you find it convinient. Here's the link: http://stackoverflow.com/questions/111102/how-do-javascript-closures-work

**Comment by tomhuhges** [comment]
thanks @Danoso89 that's a great link. have added

**Pull request merged**